### DISCUSSION TITLE
Long path - [CommunityToolkit.Maui.SourceGenerators] build error

### DISCUSSION THREAD

**Discussion by tutkus** [CommunityToolkit/Maui#399] [General]
Hi All,

That's first time when I was trying to built a solution. 
All projects except CommunityToolkit.Maui.UnitTests and CommunityToolkit.Maui.Sample were built without error however for these two I had an below:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0016	Could not write to output file '(...)\MauiCommunityToolkit\src\CommunityToolkit.Maui.UnitTests\obj\GF\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_397bb425-ad16-4165-a450-f0948b91f580.g.shared.cs' -- 'Could not find a part of the path '(...)\MauiCommunityToolkit\src\CommunityToolkit.Maui.UnitTests\obj\GF\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_397bb425-ad16-4165-a450-f0948b91f580.g.shared.cs'.'	CommunityToolkit.Maui.UnitTests (net6.0)	(...)\MauiCommunityToolkit\src\CommunityToolkit.Maui.UnitTests\CSC	1	Active
```

Checking it a bit deeper I figured out that the problem is with my Windows10 workstation. To be more specyfic it was related to LongPath which by default is disabled so in my cases becasue MAUI project were put a bit deeper in the project structure I had that issue.

The solution for that issue is quite strightforward.
The LongPath fetaure in Windows 10 has to be enabled by adding the right registry key (if that is not present yet): More details here -> https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd
After that step the only what's left is to restart the system and build project. In my case that solved the problem.

**And now the most important:
I think it would be good to put that tips for other devs which might have the same problem like me. 
Maybe it would be good to have a section in the Wiki where that kind of tips could be pointed so everyone could easy check and find a solution.**

**Comment by bijington** [comment]
@tutkus thank you for posting this. We have had a few people that have reported the same issue and we have resolved it via comments on those issues but this should hopefully make it easier for people to discover. We could possibly look to include it in the `readme.md` file although this discussion might be enough

**Reply by bijington** [reply]
I suggest that we add a Troubleshooting section to the docs site and include the details on the long path issue and links to overcome it.

**Reply by bakerhillpins** [reply]
I'm going to suggest that you use [.NET Attributes](https://docs.microsoft.com/en-us/dotnet/api/system.codedom.compiler.generatedcodeattribute?view=net-6.0) to identify the generator responsible rather than directory structure on disk:

In my Source Generator code I inject information about the Generator that created the file using attributes such as: (I'm using Scriban templates here and can provide an example if you'd like)

```
namespace {{ RootNamespace }}
{
    [ GeneratedCode("{{ GeneratorName }}", "{{ Version }}" ) ]
    [ CompilerGenerated ]
    internal class AGeneratedClass : ISomeInterface
```

I really don't understand the need for the extra layer of indirection beyond simply adding a `\CommunityToolkit.SourceGenerators\` sub directory with namespace prefixed filenames?

**Comment by CliffAgius** [comment]
Closed as answered
